Source/WebCore/page/RemoteDOMWindow.cpp

```diff
@@ -113,13 +114,41 @@ WindowProxy* RemoteDOMWindow::parent() const
     return &m_frame->windowProxy();
 }
 
-void RemoteDOMWindow::postMessage(JSC::JSGlobalObject&, LocalDOMWindow& incumbentWindow, JSC::JSValue message, const String& targetOrigin, Vector<JSC::Strong<JSC::JSObject>>&&)
-{
-    UNUSED_PARAM(incumbentWindow);
-    UNUSED_PARAM(message);
-    UNUSED_PARAM(targetOrigin);
 
-    // FIXME: Implemented this.
+void RemoteDOMWindow::postMessage(JSC::JSGlobalObject& lexicalGlobalObject, LocalDOMWindow& incumbentWindow, JSC::JSValue message, const String& targetOrigin, Vector<JSC::Strong<JSC::JSObject>>&& transfer)
+{
+    RefPtr sourceDocument = incumbentWindow.document();
+    // Compute the target origin. We need to do this synchronously in order
+    // to generate the SyntaxError exception correctly.
+    std::optional<SecurityOriginData> target;
+    if (targetOrigin == "/"_s) {
+        if (!sourceDocument)
+            return;
+        target = sourceDocument->securityOrigin().data();
+    } else if (targetOrigin != "*"_s) {
```

Comment:
I feel like this can share some code with localdomwindow::postMessage by moving some code to DOMWindow.cpp

Reply:
there seem to be subtle differences (like raising exception in case of LocalDOMWindow::postMessage, which makes me think that trying to share code here will result in unnecessary complication. Let me know if you strongly prefer that.

Reply:
This should raise the same exceptions.  Any differences should be related to the remoteness.  In any case, this is a lot of duplicate code that should be refactored in an elegant way to make it so we don't have duplicate code.  We don't want to make a change in one and forget to make the same change in the other.